src/dotnet/APIView/APIViewWeb/Repositories/GitHubArtifactRepository.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace APIViewWeb.Repositories
+{
+    public class GitHubArtifactRepository : IArtifactRepository
+    {
+        private readonly IConfiguration _configuration;
+
+        public GitHubArtifactRepository(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public async Task<Stream> DownloadPackageArtifact(
+            string repoName, string buildId, string artifactName, string filePath, string project, string format = "file")
+        {
+            using (var client = new HttpClient())
+            {
+                var gitHubToken = _configuration["github-access-token"];
```

Comment:
@praveenkuttappan as we talked we will want to come up with a plan to refresh this token in cases where we rotate.